essentials-framework/Essentials Core/PepperDashEssentialsBase/Secrets/CrestronSecretsProvider.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using Crestron.SimplSharp;
+using Crestron.SimplSharp.CrestronDataStore;
+using PepperDash.Core;
+
+
+namespace PepperDash.Essentials.Core
+{
+    public class CrestronSecretsProvider : ISecretProvider
+    {
+        public string Key { get; set; }
+        //Added for reference
+        //private readonly bool _secureSupported;
+        public CrestronSecretsProvider(string key)
+        {
+            Key = key;
+            //Added for future encrypted reference
+            //_secureSupported = CrestronSecureStorage.Supported;
+
+            //if (_secureSupported)
+            //{
+            //      return;
+            //}
+            CrestronDataStoreStatic.InitCrestronDataStore();

```

Comment:
Does this need to happen for every instance of this provider? Not that we have a factory for it, but it is conceivable to have multiple instances. It might make more sense to put this in a static constructor, that way it only happens once, not every time a new instance is created.